### ISSUE TITLE
Empty result with a query (where, whereArgs) on Android

### ISSUE THREAD

**Issue by riccardoratta** [tekartik/sqflite#64]
My code has the following query:
```
var result = await (await _connection).query(
    'recordings',
    columns: ['id', 'content', 'file', 'speaker', 'reference'],
    where: 'speaker = ?',
    whereArgs: [1]);

print('result: $result');
```
The (wrong) output from sqflite in Android is:
```
D/Sqflite (19920): [Thread[Sqflite,5,main]] SELECT id, content, file, speaker, reference FROM recordings WHERE speaker = ? [1]
I/flutter (21657): result: []
```
But the same query on the database ([database.sqlite.zip](https://github.com/tekartik/sqflite/files/2077028/database.sqlite.zip), tested with TablePlus) return 2 speakers.

The result on iOS (Simulator) is correct.

Can someone help me? Am I doing something wrong?

**Comment by alextekartik** [comment]
(I put your db in the sample code for future test until I manage to find a way to reproduce it without it, let me know if you want me to remove it)

**Comment by riccardoratta** [comment]
@alextekartik no problem, you can use the database!

**Comment by alextekartik** [comment]
Fixed in 0.10.0-dev.2 (actually paving the way for a 1.0 version so you might see some deprecated features...). It is definitely an issue on the Android API, converting all parameters to String that does not work with a definition like yours:
```
CREATE TABLE recordings (
      id INTEGER PRIMARY KEY AUTOINCREMENT,
      content TEXT NOT NULL,
      file TEXT NOT NULL,
      speaker REFERENCES speakers(id) ON DELETE CASCADE,
      reference INTEGER
    );
```

but that would work with
```
PRAGMA foreign_keys=ON
CREATE TABLE recordings (
      id INTEGER PRIMARY KEY AUTOINCREMENT,
      content TEXT NOT NULL,
      file TEXT NOT NULL,
      speaker INTEGER,
      FOREIGN_KEY(speaker) REFERENCES speakers(id) ON DELETE CASCADE,
      reference INTEGER
    );
```

let me know if that solves your issue
